src/platform/Darwin/ConfigurationManagerImpl.cpp

```diff
@@ -137,14 +138,55 @@ ConfigurationManagerImpl & ConfigurationManagerImpl::GetDefaultInstance()
 
 CHIP_ERROR ConfigurationManagerImpl::Init()
 {
-    CHIP_ERROR err;
-
+#if CHIP_DEVICE_CONFIG_ENABLE_PERSISTENT_STORAGE
     // Initialize the generic implementation base class.
-    err = Internal::GenericConfigurationManagerImpl<PosixConfig>::Init();
-    SuccessOrExit(err);
+    ReturnErrorOnFailure(Internal::GenericConfigurationManagerImpl<PosixConfig>::Init());
 
-exit:
-    return err;
+    ReturnErrorOnFailure(DeviceLayer::PersistedStorage::KeyValueStoreMgrImpl().Init("chip.store"));
+
+    uint32_t rebootCount;
+    if (!PosixConfig::ConfigValueExists(PosixConfig::kCounterKey_RebootCount))
+    {
+        // The first boot after factory reset of the Node.
+        ReturnErrorOnFailure(StoreRebootCount(1));
+    }
+    else
+    {
+        ReturnErrorOnFailure(GetRebootCount(rebootCount));
+        ReturnErrorOnFailure(StoreRebootCount(rebootCount + 1));
+    }
+
+    if (!PosixConfig::ConfigValueExists(PosixConfig::kCounterKey_TotalOperationalHours))
+    {
+        ReturnErrorOnFailure(StoreTotalOperationalHours(0));
+    }
+
+    if (!PosixConfig::ConfigValueExists(PosixConfig::kCounterKey_BootReason))
+    {
+        ReturnErrorOnFailure(StoreBootReason(DiagnosticDataProvider::BootReasonType::Unspecified));
+    }
+
+    if (!PosixConfig::ConfigValueExists(PosixConfig::kConfigKey_RegulatoryLocation))
+    {
+        uint32_t location = to_underlying(chip::app::Clusters::GeneralCommissioning::RegulatoryLocationType::kIndoor);
+        ReturnErrorOnFailure(WriteConfigValue(PosixConfig::kConfigKey_RegulatoryLocation, location));
+    }
+
+    if (!PosixConfig::ConfigValueExists(PosixConfig::kConfigKey_LocationCapability))
+    {
+        uint32_t location = to_underlying(chip::app::Clusters::GeneralCommissioning::RegulatoryLocationType::kIndoor);
+        ReturnErrorOnFailure(WriteConfigValue(PosixConfig::kConfigKey_LocationCapability, location));
+    }
+
+    // If the fail-safe was armed when the device last shutdown, initiate a factory reset.
+    bool failSafeArmed;
+    if (GetFailSafeArmed(failSafeArmed) == CHIP_NO_ERROR && failSafeArmed)
+    {
+        ChipLogProgress(DeviceLayer, "Detected fail-safe armed on reboot; initiating factory reset");
+        InitiateFactoryReset();
+    }
```

Comment:
This is not spec-compliant (neither today, but certainly not to cut and paste the bug some more). We should not factory reset the entire device, but only revert the last partially configured state. In fact, if the fail-safe was just for network reconfiguration, and AddNOC/CSRRequest were never called, there is no factory reset of a fabric to do on Fail-safe expiry

Reply:
That makes perfect sense. I will just remove this block from the Darwin bits.